sanic_routing/tree.py

```diff
@@ -57,186 +88,230 @@ def display(self) -> None:
+                and self.groups
+                and not self.first
+                and first_sibling
+            ):
+                self.equality_check = first_sibling.equality_check
+
+            # Maybe try and sneak an equality check in?
+            if_stmt = "if"
             len_check = (
                 f" and num == {self.level}"
-                if not self.children and not equality_check
+                if not self.children and not self.equality_check
                 else ""
             )
+
+            self.equality_check = self.equality_check or bool(len_check)
+
             src.append(
                 Line(
-                    f'{if_stmt} parts[{level}] == "{self.part}"{len_check}:',
-                    indent + 1,
+                    f'{if_stmt} parts[{idx}] == "{self.part}"{len_check}:'
+                    "  # CHECK 4",
+                    indent,
                 )
             )
-            if self.children:
-                return_bump += 1
+            self.base_indent += 1
 
-        if self.group:
-            location = delayed if self.children else src
+        # Get ready to return some handlers
+        if self.groups:
+            return_indent = indent + return_bump
             route_idx: t.Union[int, str] = 0
-            if self.group.requirements:
-                route_idx = "route_idx"
-                self._inject_requirements(
-                    location, indent + return_bump + bool(not self.children)
-                )
-            if self.group.params and not self.group.regex:
-                if not self.last:
-                    return_bump += 1
-                if self.parent.has_deferred:
-                    return_bump += 1
-                self._inject_params(
-                    location,
-                    indent + return_bump + bool(not self.children),
-                    not self.parent.children_param_injected,
-                )
-                if not self.parent.children_param_injected:
-                    self.parent.children_param_injected = True
-                if self._has_nested_regex(self):
-                    return_bump += 1
-            param_offset = bool(self.group.params)
-            return_indent = (
-                indent + return_bump + bool(not self.children) + param_offset
-            )
-            if self.group.regex:
-                if self._has_nested_path(self, recursive=False):
-                    location.append(Line("...", return_indent - 1))
-                    return_indent = 2
-                    location = final
-                    self._mark_parents_defer(self.parent)
-                self._inject_regex(
-                    location,
-                    return_indent,
-                    not self.parent.children_param_injected,
+            location = delayed
+
+            # Do any missing equality_check
+            if not self.equality_check:
+                conditional = "elif" if self.children else "if"
```

Comment:
Why is this `elif` when there are children? Maybe add a clarifying comment (or better yet, refactor if vs. elif management to a code generation class, if we always have a sequence of `if` followed only by `elif`s within a block).

Reply:
added
```
# If if we have not done an equality check and there are
# children nodes, then we know there is a CHECK 1
# for the children that start at the same level, and will
# be an exclusive conditional to what is being evaluated here.
# Therefore, we can use elif
#     example:
#         if num == 7:  # CHECK 1
#             child_node_stuff
#         elif num == 6:  # CHECK 5
#             current_node_stuff
```